### ISSUE TITLE
Multiple orders per user per event

### ISSUE THREAD

**Issue by harrislapiroff** [littleweaver/django-brambling#260]
I am still able to edit the attendee on a pass after it has been refunded.

Also, I'd prefer that refunded passes don't continue to show up on new orders.

![screenshot 2014-12-05 13 50 15](https://cloud.githubusercontent.com/assets/296631/5321484/ac888df6-7c89-11e4-8a68-948c80acdd77.png)

**Comment by melinath** [comment]
Waiting on pull request #303 to land before continuing on this.

**Comment by melinath** [comment]
@nmorduch @harrislapiroff I'd like to get this done soonish, but it'll need one of you to do the summary page redesign. When might that be in the cards?

**Comment by melinath** [comment]
This will actually require a database change, potentially. Right now, the Order model is used both for recording payments and also as the go-to place to find whether a person has registered for an event. I.e. the order code *is* the person's information for the event. It's intended to be an option for how people go about picking up their registration packets (as opposed to by name). That doesn't work if there's multiple orders. So we need a sub-model of some kind. I'd suggest either Cart or Receipt. Thoughts?

**Comment by nmorduch** [comment]
I think it's actually a little weird to have one order code that can go for multiple people.
What if you have an attendee code instead, to mark you as having registered?

**Comment by melinath** [comment]
That is also a thing we could do, but it would be a separate ticket. *This* ticket is about having multiple "rounds" of purchasing rather than lumping all the purchase / refund data into one bucket.

**Comment by melinath** [comment]
Though the order code is also nice because it gives anonymous users a link to their order info.

**Comment by nmorduch** [comment]
I wrote a whole thing and then I think I realized that basically rather than having Orders and Carts, I think you should have Accounts and Orders. And you can have an "account" even if you're an anonymous registrant.

**Comment by melinath** [comment]
So `Account` would essentially be what `Order` is now?

**Comment by nmorduch** [comment]
I'm not sure I understand what `Order` is well enough to answer that. But probably? Except the `Account` should also be the same as like literal accounts as we already have.

**Comment by melinath** [comment]
@nmorduch It sounds like you're saying that Account would be what people use to log in. Is that accurate? Or just similar to that?

**Comment by nmorduch** [comment]
Yeah. But right now they're pretty closely tied anyway, so I don't know that it would cause problems to give some of the `Order`'s function to `Account`

**Comment by melinath** [comment]
@nmorduch It might help if I explain the function of the involved models.

`Person` is our user model. It handles authentication and site-global data storage.
`Event` is, of course, the model that handles data related to events.
`Order` is (currently) an accounting of the relationship between an Event and a Person. So it holds the registration information for that person for that event. Some of this is global to the event (such as who the person is, if there is a person attached) and some of this is partitioned off into other models (Attendees, Payments, Refunds, etc.)

The issue then is that every time you check out, it would be useful to keep a record of what happened with that checkout so that the user can go back and look at that individually. So we still need a model to encapsulate the relationship between a user and an Event (which is what Order does right now) but we also need a separate model to keep track of what was bought when and why.

Hmm... here's a thought. Maybe we could just add an M2M field from Transaction to BoughtItem? If transactions were self-aware about what items were related to them, it would give more information to event organizers and it would allow for receipt-style display of items.

**Comment by nmorduch** [comment]
I guess I think that the relationship between a user and an Event should be determined by attendees? 

But your last solution sounds sensible not-terribly-inelegant.

**Comment by melinath** [comment]
Isn't this resolved now? The self-aware transactions landed. If there are specific other issues, I think they should be split into other tickets.

**Comment by nmorduch** [comment]
I think you're probably right.

**Comment by harrislapiroff** [comment]
Is this resolved? The primary reason I wanted multiple orders is because I didn't want to see items from previous orders while I was in the process of ordering something new because that's confusing. Maybe that's a separate issue, but it definitely has not been fixed.

**Comment by harrislapiroff** [comment]
And also to be able to access each order as a separate receipt, basically.

**Comment by nmorduch** [comment]
Then we probably want issues for those things.

**Comment by harrislapiroff** [comment]
I guess I'm confused because I thought that's what this ticket was and it doesn't seem resolved to me.

**Comment by nmorduch** [comment]
Specifically:
* Differentiate in UI btw prev and current orders
* Provide user w/ list of orders they have made

**Comment by harrislapiroff** [comment]
see comment earlier in the thread: https://github.com/littleweaver/django-brambling/issues/260#issuecomment-67785983

**Comment by harrislapiroff** [comment]
I'll create separate tickets, but unless people really disagree with me, I'd like to see those issues be given some attention, because I thought that was what was getting resolved here.

**Comment by nmorduch** [comment]
I think that this was just too big an issue after we discussed it a lot. So part of it got fixed and part of it didn't. Separate issues, now that we have capability to address them, will be helpful.

**Comment by melinath** [comment]
@harrislapiroff Those got lost because the discussion spiraled off that topic. I agree they're important. That's why they need separate tickets.

**Comment by nmorduch** [comment]
But yes, we'll give them attention <3

**Comment by harrislapiroff** [comment]
Sorry I'm being grumpy about this. Those features were the exclusive reason I created this ticket, so it's irritating and surprising to me that they got lost in the shuffle. But I know I'm being unreasonable and I can see why that happened.

**Comment by nmorduch** [comment]
I'm going to make an issue specifically about the summary page, because I think that was part of what I wanted to do with it before and forgot.
